Core/include/Acts/Vertexing/SeedVertexFinder.hpp

```diff
@@ -0,0 +1,193 @@
+
+  /// Const access to the config
+  const Config& config() const { return m_cfg; }
+
+  /// @brief Constructor
+  ///
+  /// @param cfg Configuration object
+  /// @param lgr Logging instance
+  SeedVertexFinder(const Config& cfg,
+                   std::unique_ptr<const Logger> lgr =
+                       getDefaultLogger("SeedVertexFinder", Logging::DEBUG));
+
+  /// @brief Destructor
+  ~SeedVertexFinder() = default;
+
+  /// @brief Finds the vertex based on the provided spacepoints
+  /// @param spacepoints Vector of the input spacepoints; they do not need to be sorted anyhow
+  /// @return Position of the vertex along z-axis
+  Acts::Vector3 findVertex(const std::vector<spacepoint_t>& spacepoints);
+
+ private:
+  /// @brief Struct to store spacepoint combinations from near, middle, and far parts of the detector
+  struct Triplet {
+    Triplet(const spacepoint_t& aa, const spacepoint_t& bb,
+            const spacepoint_t& cc)
+        : a(aa),
+          b(bb),
+          c(cc),
+          ray(Acts::Vector3::Zero(), Acts::Vector3::Zero()) {}
+    const spacepoint_t &a, &b, &c;
+
+    Acts::Ray3D ray;
+  };
+
+  /// Configuration instance
+  Config m_cfg;
+
+  /// @brief Sorts spacepoints into a separate vectors for near, middle, and far spacepoints; and for each slice of phi
+  /// @param spacepoints Vector of the input spacepoints;
+  /// @return Vector of vectors for each set of spacepoints
+  std::vector<std::vector<std::vector<
+      std::vector<std::pair<spacepoint_t const*, Acts::ActsScalar>>>>>
+  sortSpacepoints(const std::vector<spacepoint_t>& spacepoints) const;
+
+  /// @brief
+  /// @param a
+  /// @param b
+  /// @return
+  static bool compareSpacepoints(
+      std::pair<spacepoint_t const*, Acts::ActsScalar>& a,
+      std::pair<spacepoint_t const*, Acts::ActsScalar>& b);
+
+  /// @brief Makes triplets from the provided vectors of near, middle, and far spacepoints; and for each slice of phi
+  /// @param sorted_spacepoints Vector of input vector of vectors for each set of spacepoints
+  /// @return Vector of valid triplets
+  std::vector<Triplet> findTriplets(
+      const std::vector<std::vector<std::vector<
+          std::vector<std::pair<spacepoint_t const*, Acts::ActsScalar>>>>>&
+          sorted_spacepoints);
+
+  /// @brief Validate the triplet based on "maxXYdeviation", "maxXYZdeviation", "maxZPosition", and "maxRPosition"
+  /// @param triplet A single triplet to be validated
+  /// @return True if the deviations and fitted ray are within the configured ranges
+  ///         If "minimalizeWRT"=="rays", then the fitted ray is also saved to
+  ///         the triplet for later
+  bool isTripletValid(Triplet& triplet);
+
+  /// @brief Calculates equation of the plane (alpha*x + beta*y + gamma*z + delta = 0), given the three points
+  /// @param triplet A single triplet (with 3 spacepoints)
+  /// @return A pair of {{alpha,beta,gamma},delta}
+  std::pair<Acts::Vector3, Acts::ActsScalar> makePlaneFromTriplet(
```

Comment:
Up!

Reply:
Hi @andiwand and @asalzburger,
what is the motivation to make it `static`? The method is used only within this class and I don't think it would be useful elsewhere. I get the motivation to make them `const` (which I overlooked for some of them) but I don't understand what we'll gain with `static` methods?
Also, what do you mean by "Up!"? I sorted the methods by the order they are used. Thus, first is sorting spacepoints, then finding triplets, then finding the vertex.

Thank you, Petr

Reply:
hey @pbalek ! this method depends on the state of the algorithm i.e. the config, right? I think this is what I later discovered during review.

In general I would say that all methods that do not depend on the object state should be static to explicitly tell this fact.

I think `Up!` just meant :+1: to my suggestion (which does not seem to apply anyways) :slightly_smiling_face:

Reply:
Hi @andiwand,
> In general I would say that all methods that do not depend on the object state should be static to explicitly tell this fact.

Ah, this makes sense! Thank you, I'll change them where applicable.

Cheers, Petr